apps/server/src/modules/fwu-learning-contents/controller/api-test/fwu-learning-contents.api.spec.ts

```diff
@@ -24,134 +22,108 @@ class API {
 	}
 }
 
-const createRndInt = (max) => Math.floor(Math.random() * max);
-type UploadFile = { Key: string; Body: string | Buffer; ContentType: string };
-
-/**
- * Creates a local S3 instance and uploads two example files
- */
-async function setup(config: S3Config) {
-	const client = new S3Client({
-		region: config.region,
-		credentials: {
-			accessKeyId: config.accessKeyId,
-			secretAccessKey: config.secretAccessKey,
-		},
-		endpoint: config.endpoint,
-		forcePathStyle: true,
-		tls: true,
-	});
-
-	const exampleTextFile = {
-		Key: '12345/example.txt',
-		Body: 'example-content',
-		ContentType: 'text/plain',
-	};
-
-	const exampleBinaryFile = {
-		Key: 'example',
-		Body: Buffer.from([1, 2, 3, 4]),
-		ContentType: 'application/octet-stream',
-	};
-
-	const upload = async (file: UploadFile) =>
-		new Upload({
-			client,
-			params: {
-				Bucket: config.bucket,
-				...file,
-			},
-		}).done();
-
-	await upload(exampleTextFile);
-	await upload(exampleBinaryFile);
-
-	return { exampleTextFile, exampleBinaryFile, client };
-}
-const port = 10000 + createRndInt(10000);
-const overriddenS3Config = Object.assign(s3Config, { endpoint: `http://localhost:${port}` });
-const s3instance = new S3rver({
-	directory: `/tmp/s3rver_test_directory${port}`,
-	resetOnClose: true,
-	port,
-	configureBuckets: [{ name: s3Config.bucket, configs: [] }],
-});
-
 describe('FwuLearningContents Controller (api)', () => {
 	let app: INestApplication;
 	let api: API;
+	let s3ClientAdapter: DeepMocked<S3ClientAdapter>;
 
 	beforeAll(async () => {
-		await s3instance.run();
 		const module = await Test.createTestingModule({
 			imports: [FwuLearningContentsTestModule],
-			providers: [
-				FwuLearningContentsTestModule,
-				{
-					provide: 'S3_Config',
-					useValue: overriddenS3Config,
-				},
-			],
 		})
 			.overrideGuard(JwtAuthGuard)
 			.useValue({
 				canActivate() {
 					return true;
 				},
 			})
+			.overrideProvider(S3ClientAdapter)
+			.useValue(createMock<S3ClientAdapter>())
 			.compile();
 
 		app = module.createNestApplication();
 		await app.init();
+		s3ClientAdapter = module.get(S3ClientAdapter);
 
 		api = new API(app);
 	});
 
 	afterAll(async () => {
 		await app.close();
-		await s3instance.close();
 	});
 
 	describe('requestFwuContent', () => {
 		Configuration.set('FEATURE_FWU_CONTENT_ENABLED', true);
-		let exampleTextFile: UploadFile;
 		describe('when the file has a file-extension', () => {
+			const setup = () => {
+				const fileKey = '12345/example.txt';
```

Comment:
I'm wondering about the name "fileKey". It looks like a path. Key sounds weird... (I saw it was named that way before, too.)

Reply:
Just saw that in the controller it is named "path".